pkgs/stdenv/generic/setup.sh

```diff
@@ -1075,7 +1075,7 @@ buildPhase() {
         # Old bash empty array hack
         # shellcheck disable=SC2086
         local flagsArray=(
-            ${enableParallelBuilding:+-j${NIX_BUILD_CORES} -l${NIX_BUILD_CORES}}
```

Comment:
A constant value would not make sense here IMHO. Which value makes rather depends on the system and its configuration (i.e. number of cores and nix's max-jobs/cores): A value of 2.0 may be a good choice for a laptop but not for a big server with lots of CPU cores.
If max-jobs is set to the number of cores in the system, it could make sense to set `-l <number cores>` to avoid overly high system loads.
I would be interesting to know the motivation why `-l${NIX_BUILD_CORES}` was set here in the first place. Maybe @Ericson2314 knows more?

Reply:
It is good to have safeguards in place. However, when the safeguards cause my builds to run with the equivalent of `-j1` on a 64-core machine, it is no longer feasible to use Nix in any way in a professional context.

Reply:
> cause my builds to run with the equivalent of -j1 on a 64-core machine

Are you sure? I would imagine it starting out with 64 jobs and if/when system load > 64, then new jobs are delayed until load falls below 64. But that'd mean it should run the overall build with >>1 job.

It would be cool to visualize it :-)

Reply:
I am not able to run builds with `cores = 64` because the OOM killer is invoked:

![Screen Shot 2022-07-15 at 06 05 27](https://user-images.githubusercontent.com/543423/179202089-0942f6b6-5769-4664-9c57-ade525893446.png)

If I run the builds with `cores = 8`, `-j8 -l8` will be passed to `make`. This is not good because the system has a load average higher than 8, which causes the builds to slow to a crawl.

Reply:
> Isn't `-l${NIX_BUILD_CORES}` a good protection against overloading/DoS-ing build machines? Maybe it could be configurable at runtime, but I personally think the default is good.

It is certainly a protection against overloading. However, the question if it is an efficient protection. The default may be good for the main Hydra build farm. On servers with mixed load, this default does not work not work well: E.g.: on a 48 core machine which dedicates half of its cores to a constant, non-build load, and the other half to nix-build jobs, this results in the named problem of gross under utilization. Running nix with `-l24` will not result in the desired result of using 24 cores for the build job, but in nix (or `make`) to only use a single core.

Reply:
@centromere @markuskowa: Good point.